operator/v1/types_network.go

```diff
@@ -100,12 +100,79 @@ type DefaultNetworkDefinition struct {
 	KuryrConfig *KuryrConfig `json:"kuryrConfig,omitempty"`
 }
 
+// SimpleMacvlanConfig contains configurations for macvlan interface.
+type SimpleMacvlanConfig struct {
+	// master is the host interface to create the macvlan interface from.
+	// If not specified, it will be default route interface
+	// +optional
+	Master string `json:"master,omitempty"`
+
+	// IPAMConfig configures IPAM module will be used for IP Address Management(IPAM).
+	IPAMConfig IPAMConfig `json:"ipamConfig"`
+
+	// mode is the macvlan mode: bridge, private, vepa, passthru. The default is bridge
+	// +optional
+	Mode MacvlanMode `json:"mode,omitempty"`
+
+	// mtu is the mtu to use for the macvlan interface. if unset, host's
+	// kernel will select the value.
+	// +optional
+	MTU uint32 `json:"mtu,omitempty"`
+}
+
+// StaticIPAMAddresses provides IP address/Gateway pairs for static IPAM addresses
+type StaticIPAMAddresses struct {
+	// Address is the ip address in CIDR
+	Address string `json:"address,omitempty"`
+	// Gateway is ip inside of subnet to designate as the gateway
+	Gateway string `json:"gateway,omitempty"`
```

Comment:
If `omitempty` is correct here then you need to say what the behavior when it's omitted is.

Reply:
I added for Address. In gateway case, just no gateway for the subnet, hence it is trivial, hence not added.